toml.py

```diff
@@ -696,38 +696,45 @@ def _dump_sections(o, sup):
     retstr += arraystr
     return (retstr, retdict)
 
+
 def _dump_value(v):
-    if isinstance(v, list):
-        t = []
-        retval = "["
-        for u in v:
-            t.append(_dump_value(u))
-        while t != []:
-            s = []
-            for u in t:
-                if isinstance(u, list):
-                    for r in u:
-                        s.append(r)
-                else:
-                    retval += " " + str(u) + ","
-            t = s
-        retval += "]"
-        return retval
-    if isinstance(v, (str, unicode)):
-        v = "%r" % v
-        if v[0] == 'u':
-            v = v[1:]
-        singlequote = v[0] == "'"
-        v = v[1:-1]
-        if singlequote:
-            v = v.replace("\\'", "'")
-            v = v.replace('"', '\\"')
-        v = v.replace("\\x", "\\u00")
-        return str('"'+v+'"')
-    if isinstance(v, bool):
-        return str(v).lower()
-    if isinstance(v, datetime.datetime):
-        return v.isoformat()[:19]+'Z'
-    if isinstance(v, float):
-        return str(v)
-    return v
+    dump_value = {
```

Comment:
The way that this is constructed, it calls every single possibility on `v` when it should only evaluate the matching possibility. I think that `dump_value` should map types to functions and the `return` statement below should call that function on `v` (with the default being an identity lambda, something like `lambda x: x`).